isis/tests/AngleTest.cpp

```diff
@@ -0,0 +1,304 @@
+   }
+   catch(...) {
+     FAIL() << "Expected IException: **PROGRAMMER ERROR** Cannot compare a invalid angles with the < operator, but got something else "; 
+   }
+}
+
+TEST(AngleExceptions, NullAngleLessThanOrEqual){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+
+  try {
+    Isis::Angle() <= angle1;
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the < operator." );
+   }
+   catch(...) {
+     FAIL() << "Expected IException: **PROGRAMMER ERROR** Cannot compare a invalid angles with the < operator, but got something else "; 
+   }
+}
+
+
+TEST(AngleExceptions, GreaterThanNullAngle){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+  
+  try {
+    angle1 > Isis::Angle();
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the > operator." );
+   }
+   catch(...) {
+     FAIL() << "Expected IException: **PROGRAMMER ERROR** Cannot compare a invalid angles with the > operator, but got something else "; 
+   }
+}
+
+
+TEST(AngleExceptions, GreaterThanOrEqualToNullAngle){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+  
+  try {
+    angle1 >= Isis::Angle();
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the > operator." );
+   }
+   catch(...) {
+     FAIL() << "Expected IException: **PROGRAMMER ERROR** Cannot compare a invalid angles with the > operator, but got something else "; 
+   }
+}
+
+
+TEST(AngleOperators, Multiplication) {
+  Isis::Angle angle(30.0, Isis::Angle::Degrees );
+  angle = 2.0 * angle; 
+  EXPECT_NEAR(angle.degrees(), 60.0, TEST_EPSILON); 
+}
+
+
+TEST(Angle, wrapValue){
+  Isis::Angle angle(30., Isis::Angle::Degrees);
+  EXPECT_EQ(angle.unitWrapValue(Isis::Angle::Degrees), 360);
+  EXPECT_NEAR(angle.unitWrapValue(Isis::Angle::Radians), 6.28318531, TEST_EPSILON);
+}
+
+
+TEST(Angle, setAngleDegrees){
+  Isis::Angle angle(0.0, Isis::Angle::Degrees);
+  angle.setAngle(60.0, Isis::Angle::Degrees);
+  EXPECT_NEAR(angle.degrees(), 60.0, TEST_EPSILON);
+}
+
+
+TEST(Angle, setAngleRadians){
+  Isis::Angle angle(0.0, Isis::Angle::Radians);
+  angle.setAngle(0.5, Isis::Angle::Radians);
+  EXPECT_NEAR(angle.radians(), 0.5, TEST_EPSILON);
+}
+
+
+TEST(Angle, QStringConstructor){
+  Isis::Angle angle(QString("-70 15 30.125"));
+  EXPECT_STREQ(angle.toString().toLatin1(), "-70.258368055556 degrees");
+  Isis::Angle angle2(QString("  +70  30 11     "));
+  EXPECT_STREQ(angle2.toString().toLatin1(), "70.503055555556 degrees");
+  Isis::Angle angle3(QString("100 00 00"));
+  EXPECT_STREQ(angle3.toString().toLatin1(), "100.0 degrees"); 
```

Comment:
And here

Reply:
Depending on how @jessemapel  and @kberryUSGS resolve the string containing FP comments, it might be good to update documentation to describe a consistent method for testing this.  I err on the side of wanting to avoid any kind of string comparison that contains a FP number due to the challenges it introduces.

Reply:
@jlaura @jessemapel If possible, I like the idea of testing numerical values with many decimal places and string and/or debug output separately. In cases where we are testing the string output, we could use a very simple number like "1".

Reply:
:+1: